app/models/reservation.rb

```diff
@@ -19,6 +19,14 @@ class Reservation < ActiveRecord::Base
   validate :status_final_state
   validate :not_in_past, :available, :check_banned, on: :create
 
+  # conditional counter cache for overdue reservations
+  after_update :increment_cache, if: :checked_out?
+  after_update :decrement_cache, if: :overdue
+
+  # make counter cache update on create
+  # should only ever get called in the test suite
+  after_create :test_cache, if: :overdue
+
   # correctly update the overdue flag if necessary
   before_save :update_overdue, if: :checked_out?
```

Comment:
Yea, I guess that also would have messed up with the order of things since hypothetically `increment_cache` is dependent on `update_overdue` running successfully.